handlers/java_handler/src/main/java/mercator/MavenUtils.java

```diff
@@ -79,8 +82,47 @@ public static Document getParsedExpandedPom(File pomFile) {
             return null;
         }
 
-        Document parsedPom = readFileAsDocument(resolvedPom);
-        return parsedPom;
+        try {
+            File resolvedPomFixed = removeDuplicateXMLTag(resolvedPom);
+            Document parsedPom = readFileAsDocument(resolvedPomFixed);
+            return parsedPom;
+        } catch (java.io.IOException ex) {
+            return null;
+        }
+
+    }
+
+    /**
+     * Remove duplicate XML start tag from given XML file.
+     * @param resolvedPom
+     * @return File handle having only one XML start tag
+     * @throws IOException
+     */
+    public static File removeDuplicateXMLTag(File resolvedPom) throws IOException {
+        File resolvedPomFixed;
+        resolvedPomFixed = File.createTempFile("resolvedpom-fixed", ".xml");
+        resolvedPomFixed.deleteOnExit();
+
+        Scanner sc = new Scanner(new FileInputStream(resolvedPom));
```

Comment:
I'd suggest to define the encoding / charset in the Scanner constructor:
https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html#Scanner(java.io.File,%20java.lang.String)

You assume that the XMLs use UTF-8 (that is right), so the scanner should use the same encoding as well for the code to be portable to all architectures.

Reply:
@tisnik  Added UTF-8 encoding to Scanner now. Thanks for pointing it out.